src/Umbraco.Web.UI.Client/src/packages/core/auth/modals/umb-app-auth-modal.element.ts

```diff
@@ -93,17 +107,34 @@ export class UmbAppAuthModalElement extends UmbModalBaseElement<UmbModalAppAuthC
 						${this.data?.userLoginState === 'timedOut'
 							? html`<p style="margin-top:0">${this.localize.term('login_timeout')}</p>`
 							: ''}
-						<umb-extension-slot
-							id="providers"
-							type="authProvider"
-							default-element="umb-auth-provider-default"
-							.props=${this.props}></umb-extension-slot>
+						${when(
+							this._loading,
+							() => html`
+								<div id="loader">
+									<uui-loader></uui-loader>
+								</div>
+							`,
+							() =>
+								html` <umb-extension-slot
+									id="providers"
+									type="authProvider"
+									default-element="umb-auth-provider-default"
+									.props=${this.props}
+									.filter=${this.#filterProvider}></umb-extension-slot>`,
+						)}
 					</div>
 				</div>
 			</div>
 		`;
 	}
 
+	#filterProvider = (provider: ManifestAuthProvider) => {
+		if (!this._allowLocalLogin) {
```

Comment:
could we instead make it, that would make it easier to read IMO.

```
if (this._allowLocalLogin) {
    return true;
}
...
```

Reply:
Done :)